src/components/ContentWidgets/ImagesInRow/ImagesInRowEditor.tsx

```diff
@@ -69,18 +150,32 @@ export const ImagesInRowEditor: React.FC<WidgetEditorProps> = ({
             variant="outlined"
             color="primary"
             onClick={e => {
-                pictureCaptionPairs.push({
-                    imgLink: imgLinkText,
-                    imgCaption: imgCaptionText,
-                    extLink: extLinkText
-                });
-                setEditedContentOnChange(
-                    "imagesInRow_items",
-                    pictureCaptionPairs
-                );
-                changeImgLinkText("");
-                changeCaptionText("");
-                changeExtLinkText("");
+                if (imgLinkText === "") {
```

Comment:
For these functions it might be better to have the `if` `else` handling within the function instead of outside. That way you can do like `switchEmptyImgLinkSubmitted(imgLinkText)` on a single line instead of 5 lines for an if else. This will also let you refactor `switchE...` later on instead of coming into the render function if you think the logic has to change. (It will also potentially let us factor it to a separate file or OUTSIDE of the class).

Reply:
The rest of this stuff inside this onclick function might be better off being factored out into a function so that it is more readable.